src/gitUtils.ts

```diff
@@ -1,21 +1,43 @@
 import { exec } from "@actions/exec";
 import { execWithOutput } from "./utils";
 
-export const setupUser = async () => {
+export const setupUser = async (githubUserName: string, githubUserEmail: string) => {
   await exec("git", [
     "config",
     "--global",
     "user.name",
-    `"github-actions[bot]"`,
+    githubUserName,
   ]);
   await exec("git", [
     "config",
     "--global",
     "user.email",
-    `"github-actions[bot]@users.noreply.github.com"`,
+    githubUserEmail,
   ]);
 };
 
+export const setupCommitSigning = async (gpgPrivateKey: string) => {
+  await exec("apk", [
+    "add",
+    "--no-cache",
+    "gnupg",
+  ]);
+  await exec("gpg", ["--import"], { input: Buffer.from(gpgPrivateKey) });
+  const { stdout: keyId } = await execWithOutput(`/bin/bash -c "gpg --list-secret-keys --with-colons | grep '^sec:' | cut -d ':' -f 5"`);
```

Comment:
couldnt this be decoded from the importing's command output? something like this should display the information about what is being imported
```
gpg --with-colons --import-options show-only --import
```

the problem with listing all keys is that potentially the returned list could potentially contain many entries and we could grab some unrelated `keyId` here

Reply:
Yeah true, although the current implementation seems to be sufficient for the crowdin action, especially if consumers are just passing the one private key.

Happy to explore your solution though if this is a blocker.

Reply:
It's not a blocker per se but if this solution could be used then I would prefer using it - cause it seems to be more explicit about the whole flow. It's less dependent on other factors and the preexisting environment.

Reply:
Yeah that makes sense, I won't pick this up for a few days now but I've addressed all the other comments, if it's sufficient as is then feel free to merge but if not I'll come back to this